querybook/webapp/components/ReviewIndicatorButton/ReviewIndicatorButton.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import clsx from 'clsx';
+import { getButtonComponentByType } from 'ui/Button/Button';
+import type { IQueryReview } from 'const/queryExecution';
+import type { TooltipDirection } from 'const/tooltip';
+
+interface IReviewIndicatorButtonProps {
+    review: IQueryReview;
+    onClick: () => void;
+    tooltipPos?: TooltipDirection;
+    className?: string;
+}
+
+const STATUS_CONFIG = {
+    pending: {
+        icon: 'Clock' as const,
+        title: 'Pending Review',
+    },
+    approved: {
+        icon: 'CheckCircle' as const,
+        title: 'Review Approved',
+    },
+    rejected: {
+        icon: 'XCircle' as const,
+        title: 'Review Rejected',
+    },
+} as const;
+
+export const ReviewIndicatorButton: React.FC<IReviewIndicatorButtonProps> = ({
```

Comment:
have an expanded/hide state where the button is more clear on what it can do

when hidden, I think the button works as is now
when expanded, change the button to `Hide Review`